internal/topology/variables/util.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package variables
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// convertToUpstreamJSONSchemaProps converts a clusterv1.JSONSchemaProps to
+// apiextensions.JSONSchemaProp.
+// NOTE: This func is used whenever we want to use one of the upstream libraries, because
+// they are using apiextensions.JSONSchemaProp.
+// NOTE: If new fields are added to clusterv1.JSONSchemaProps (e.g. to support complex types), the corresponding
+// schema validation must be added to validateSchema too.
+func convertToUpstreamJSONSchemaProps(schema *clusterv1.JSONSchemaProps, _ *field.Path) (*apiextensions.JSONSchemaProps, error) {
```

Comment:
Just saw that field.Path is unused, I think we can drop it

Reply:
```suggestion
func convertToAPIextensionsJSONSchemaProps(schema *clusterv1.JSONSchemaProps, _ *field.Path) (*apiextensions.JSONSchemaProps, error) {
```

upstream is kind of "relative" 😄

Reply:
Yeah I now it was a bad name. Just didn't like the casing for the package